OpenRA.Mods.Common/Projectiles/Railgun.cs

```diff
@@ -0,0 +1,246 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2017 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation, either version 3 of
+ * the License, or (at your option) any later version. For more
+ * information, see COPYING.
+ */
+#endregion
+
+using System.Collections.Generic;
+using System.Drawing;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Projectiles
+{
+	[Desc("Laser effect with helix coiling around.")]
+	public class RailgunInfo : IProjectileInfo
+	{
+		[Desc("Damage all units hit by the beam instead of just the target?")]
+		public readonly bool DamageActorsInLine = false;
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Can this projectile be blocked when hitting actors with an IBlocksProjectiles trait.")]
+		public readonly bool Blockable = false;
+
+		[Desc("Duration of the beam and helix")]
+		public readonly int Duration = 15;
+
+		[Desc("Equivalent to sequence ZOffset. Controls Z sorting.")]
+		public readonly int ZOffset = 0;
+
+		[Desc("The width of the main trajectory. (\"beam\").")]
+		public readonly WDist BeamWidth = new WDist(86);
+
+		[Desc("The shape of the beam.  Accepts values Cylindrical or Flat.")]
+		public readonly BeamRenderableShape BeamShape = BeamRenderableShape.Cylindrical;
+
+		[Desc("Beam color in (A),R,G,B.")]
+		public readonly Color BeamColor = Color.FromArgb(128, 255, 255, 255);
+
+		[Desc("When true, this will override BeamColor parameter and draw the laser with player color."
+			+ " (Still uses BeamColor's alpha information)")]
+		public readonly bool BeamPlayerColor = false;
+
+		[Desc("Beam alpha gets + this value per tick during drawing; hence negative value makes it fade over time.")]
+		public readonly int BeamAlphaDeltaPerTick = -8;
+
+		[Desc("Thickness of the helix")]
+		public readonly int HelixThickness = 64;
```

Comment:
This and the next two appear to be a world-distances, so they should be defined as `WDist`.